app/views/unattended/provisioning_templates/snippet/preseed_kernel_options.erb

```diff
@@ -18,6 +19,11 @@ description: options for the kernel / preseed startup initialization
     options << host_param('blacklist').split(',').collect{|x| "#{x.strip}.blacklist=yes"}.join(' ')
   end
 
+  if mac
+    # hardware type is always 01 (ethernet) unless specified otherwise
+    options << "BOOTIF=#{host_param("hardware_type", "01")}-#{mac.gsub(':', '-')}"
```

Comment:
this now produces for PXEGrub2
```
linuxefi  boot/debian-mirror-RpV7E2zxrKHe-linux interface=auto url=http://foreman.example.com/unattended/provision ramdisk_size=10800 root=/dev/rd/0 rw auto BOOTIF=01-$net_default_mac BOOTIF=01-00-f0-54-1a-7e-e0 auto=true locale=en_US netcfg/disable_dhcp=false locale=en_US hostname=snapshot-ipv4-dhcp-deb10 domain=snap.example.com
```

and for iPXE:
```
kernel http://ftp.debian.org/debian/dists/wheezy/main/installer-amd64/current/images/netboot/debian-installer/amd64/linux initrd=initrd.img interface=auto url=http://foreman.example.com/unattended/provision ramdisk_size=10800 root=/dev/rd/0 rw auto BOOTIF=01-${netX/mac:hexhyp} BOOTIF=01-00-f0-54-1a-7e-e0 auto=true locale=en_US netcfg/disable_dhcp=false locale=en_US hostname=snapshot-ipv4-dhcp-deb10 domain=snap.example.com
```

see the double `BOOTIF`? that sounds wrong to me?

Reply:
yes, you are right!
I moved the BOOTIF for iPXE into the kernel options and removed the BOOTIF from both templates (iPXE and Grub2). 

As I had no issues in all my tests with BOOTIF containing the mac specified in the interface, I discarded the $net_default_mac version in case of Grub2.